ext/node/lib.rs

```diff
@@ -582,11 +523,18 @@ deno_core::extension!(deno_node,
     "zlib.ts",
   ],
   options = {
-    maybe_npm_resolver: Option<Rc<dyn NpmResolver>>,
+    maybe_npm_resolver: Option<Arc<dyn NpmResolver>>,
+    fs: Option<Arc<dyn NodeFs>>,
```

Comment:
I tried my best not to pass an `Arc` into here, but I found whatever I tried to be too unwieldy, especially with the LSP requiring Send. We can remove all the cloning that happens in the ops by changing `NodePermissions` to not have to be `&mut`.

Reply:
Unlucky, but I see the reasoning here